### DISCUSSION TITLE
how to build modules in win10 vs2017

### DISCUSSION THREAD

**Discussion by hh-kai** [baresip/baresip#2276] [Q&A]
![image](https://user-images.githubusercontent.com/19390025/196901662-0fa7689e-30b7-496d-9222-b70342d4e70e.png)

re, rem and baresip build succe
but run baresip.exe is failed
![image](https://user-images.githubusercontent.com/19390025/196902030-d13008ae-ecff-4ea9-9acf-fea68df017ae.png)

**Comment by sreimers** [comment]
You have to build static:

```cmake -B build -DSTATIC=1```

**Reply by hh-kai** [reply]
yes i build static and is default build static
![image](https://user-images.githubusercontent.com/19390025/196911025-3b8523af-535a-43f4-99cd-b4ef59cbedf3.png)
![image](https://user-images.githubusercontent.com/19390025/196910987-a009cf13-6c76-4cb0-bd91-ced75c5d1ba4.png)

**Reply by hh-kai** [reply]
![image](https://user-images.githubusercontent.com/19390025/196911165-5405e8d0-01b8-4572-b488-ba17ed55a346.png)

**Comment by sreimers** [comment]
Can't reproduce with Visual Studio 2022. Unsure if this is related to vs2017.

**Reply by hh-kai** [reply]
i try in vs2019 and get that error
![image](https://user-images.githubusercontent.com/19390025/196921772-2f7d5546-aaea-45ef-a216-e3d4be5374ac.png)
then i change CMakeLists.txt, like that
![image](https://user-images.githubusercontent.com/19390025/196922039-cc069e0c-30d0-4e9c-bee7-d386c6ba6593.png)
and pkg_config.exe is ok
![image](https://user-images.githubusercontent.com/19390025/196922319-4065aa6a-5fea-4981-a89e-78e309703002.png)

but it's useless

Any suggestions?
